### ISSUE TITLE
Importing `String` and `DateTime` fails on Windows and OSX if `PYTHONCASEOK` is set or installation directory shared between VM and host

### ISSUE THREAD

**Issue by spooning** [robotframework/robotframework#1433]
> _Originally submitted to [Google Code](http://code.google.com/p/robotframework/issues/detail?id=1433) by filip.noetzel on 13 May 2013_

I get this traceback with the latest Robot Framework 2.7.7 when executing just `pybot`.

```
Traceback (most recent call last):
  File "bin/pybot", line 4, in &lt;module&gt;
    from robot import run_cli
  File "/home/vagrant/workspace/myproject/local/lib/python2.7/site-packages/robot/**init**.py", line 22, in &lt;module&gt;
    from robot.rebot import rebot, rebot_cli
  File "/home/vagrant/workspace/myproject/local/lib/python2.7/site-packages/robot/rebot.py", line 268, in &lt;module&gt;
    from robot.conf import RebotSettings
  File "/home/vagrant/workspace/myproject/local/lib/python2.7/site-packages/robot/conf/**init**.py", line 17, in &lt;module&gt;
    from .settings import RobotSettings, RebotSettings
  File "/home/vagrant/workspace/myproject/local/lib/python2.7/site-packages/robot/conf/settings.py", line 17, in &lt;module&gt;
    from robot import utils
  File "/home/vagrant/workspace/myproject/local/lib/python2.7/site-packages/robot/utils/**init**.py", line 21, in &lt;module&gt;
    from .argumentparser import ArgumentParser
  File "/home/vagrant/workspace/myproject/local/lib/python2.7/site-packages/robot/utils/argumentparser.py", line 21, in &lt;module&gt;
    import string
  File "/home/vagrant/workspace/myproject/local/lib/python2.7/site-packages/robot/libraries/string.py", line 24, in &lt;module&gt;
    from string import ascii_lowercase, ascii_uppercase, digits
ImportError: cannot import name ascii_lowercase
```

This runs inside a virtualenv, and it seems to me that somewhere too early "robot/libraries" is prepended to sys.path, so that "robot/libraries/string.py" is imported from "robot/utils/argumentparser.py" when Python's string module (<a href="http://docs.python.org/2/library/string.html" rel="nofollow">http://docs.python.org/2/library/string.html</a>) is meant to be imported. This is probably related to absolute vs. relative imports, but I haven't figured out why and where.

I don't have the definite steps to reproduce this outside of my proprietary codebase, but will update this as soon as I find out more.

**Comment by darmenliu** [comment]
hi, how about this issue, did it has already been fixed?

**Comment by HelioGuilherme66** [comment]
The issue is closed, due to not enough info. Are you having the same problem? If so, paste here your setup and logs.

**Comment by darmenliu** [comment]
run_in_virtualenv rfcli --target CLAB12 -L trace -V variables/rcp2_reference_heat_variables.py --pythonpath ../cloudtaf/libraries -v CI_DEPLOYER_STACK_NAME:RCP20_WZC -v CI_DEPLOYER_LEAVE_STACK:True -v CI_DEPLOYER_LEAVE_IMAGE:True --test "RCP_ELAM_018 Confirm remove a non-management node by cli shall be success" testcases
[ WARN ] DEPRECATED!!: https://gitlab.fp.nsn-rdnet.net/rcp/cloudtaf libraries/RemoteScript
[ WARN ] Use cloudtaf.remotescript from https://pypi.fp.nsn-rdnet.net/crl/prod/ instead
[ WARN ] See the instructions to set up robot runner environment:
https://confluence.int.net.nokia.com/display/RCP/Robot+runner+environment+installation+instruction
[ WARN ] DEPRECATED!!: https://gitlab.fp.nsn-rdnet.net/rcp/cloudtaf libraries/RemoteScript
[ WARN ] Use cloudtaf.remotescript from https://pypi.fp.nsn-rdnet.net/crl/prod/ instead
[ WARN ] See the instructions to set up robot runner environment:
https://confluence.int.net.nokia.com/display/RCP/Robot+runner+environment+installation+instruction
[ WARN ] DEPRECATED!!: https://gitlab.fp.nsn-rdnet.net/rcp/cloudtaf libraries/RemoteScript
[ WARN ] Use cloudtaf.remotescript from https://pypi.fp.nsn-rdnet.net/crl/prod/ instead
[ WARN ] See the instructions to set up robot runner environment:
https://confluence.int.net.nokia.com/display/RCP/Robot+runner+environment+installation+instruction
[ ERROR ] Error in file '/mnt/temp/rcptaf/resources/framework/deployment/_deployment_helper.robot': Importing test library 'robot.libraries.String' failed: ImportError: cannot import name ascii_lowercase
Traceback (most recent call last):
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/String.py", line 18, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/string.py", line 18, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
PYTHONPATH:
  /mnt/temp/cloudtaf/libraries
  /mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages
  /mnt/temp/rcp_cloudtaf_virtualenv/bin
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/plat-x86_64-linux-gnu
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-old
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-dynload
  /usr/lib/python2.7
  /usr/lib/python2.7/plat-x86_64-linux-gnu
  /usr/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/site-packages
  /mnt/temp/rcptaf
  /mnt/temp/rcptaf/libraries
  /mnt/temp/rcptaf/resources
  /mnt/temp/rcptaf/testcases/ELAM/libraries
[ ERROR ] Error in file '/mnt/temp/rcptaf/resources/framework/deployment/_deployment_via_cbam.robot': Importing test library 'robot.libraries.String' failed: ImportError: cannot import name ascii_lowercase
Traceback (most recent call last):
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/String.py", line 18, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/string.py", line 18, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
PYTHONPATH:
  /mnt/temp/cloudtaf/libraries
  /mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages
  /mnt/temp/rcp_cloudtaf_virtualenv/bin
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/plat-x86_64-linux-gnu
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-old
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-dynload
  /usr/lib/python2.7
  /usr/lib/python2.7/plat-x86_64-linux-gnu
  /usr/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/site-packages
  /mnt/temp/rcptaf
  /mnt/temp/rcptaf/libraries
  /mnt/temp/rcptaf/resources
  /mnt/temp/rcptaf/testcases/ELAM/libraries
[ ERROR ] Error in file '/mnt/temp/rcptaf/resources/framework/deployment/deployment_helper.robot': Importing test library 'robot.libraries.String' failed: ImportError: cannot import name ascii_lowercase
Traceback (most recent call last):
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/String.py", line 18, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/string.py", line 18, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
PYTHONPATH:
  /mnt/temp/cloudtaf/libraries
  /mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages
  /mnt/temp/rcp_cloudtaf_virtualenv/bin
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/plat-x86_64-linux-gnu
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-old
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-dynload
  /usr/lib/python2.7
  /usr/lib/python2.7/plat-x86_64-linux-gnu
  /usr/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/site-packages
  /mnt/temp/rcptaf
  /mnt/temp/rcptaf/libraries
  /mnt/temp/rcptaf/resources
  /mnt/temp/rcptaf/testcases/ELAM/libraries
[ ERROR ] Error in file '/mnt/temp/rcptaf/resources/framework/deployment/deployment_helper.robot': Importing test library 'Cluster.clustermapper.ClusterMapper' failed: ImportError: No module named clusteragent
Traceback (most recent call last):
  File "/mnt/temp/cloudtaf/libraries/Cluster/clustermapper.py", line 5, in <module>
    from Cluster.clusteragent import ClusterAgent
PYTHONPATH:
  /mnt/temp/cloudtaf/libraries
  /mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages
  /mnt/temp/rcp_cloudtaf_virtualenv/bin
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/plat-x86_64-linux-gnu
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-old
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-dynload
  /usr/lib/python2.7
  /usr/lib/python2.7/plat-x86_64-linux-gnu
  /usr/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/site-packages
  /mnt/temp/rcptaf
  /mnt/temp/rcptaf/libraries
  /mnt/temp/rcptaf/resources
  /mnt/temp/rcptaf/testcases/ELAM/libraries
[ ERROR ] Error in file '/mnt/temp/rcptaf/resources/rcp/rcp_ci_suite_setup_definitions.robot': Importing test library 'robot.libraries.String' failed: ImportError: cannot import name ascii_lowercase
Traceback (most recent call last):
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/String.py", line 18, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/string.py", line 18, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
PYTHONPATH:
  /mnt/temp/cloudtaf/libraries
  /mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages
  /mnt/temp/rcp_cloudtaf_virtualenv/bin
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/plat-x86_64-linux-gnu
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-old
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-dynload
  /usr/lib/python2.7
  /usr/lib/python2.7/plat-x86_64-linux-gnu
  /usr/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/site-packages
  /mnt/temp/rcptaf
  /mnt/temp/rcptaf/libraries
  /mnt/temp/rcptaf/resources
  /mnt/temp/rcptaf/testcases/ELAM/libraries
[ ERROR ] Error in file '/mnt/temp/rcptaf/resources/rcp/rcp_ci_suite_setup_definitions.robot': Importing test library 'RCPClusterMapper' failed: ImportError: No module named clusteragent
Traceback (most recent call last):
  File "/mnt/temp/rcptaf/libraries/RCPClusterMapper/__init__.py", line 1, in <module>
    from rcpclustermapper import RCPClusterMapper
  File "/mnt/temp/rcptaf/libraries/RCPClusterMapper/rcpclustermapper.py", line 6, in <module>
    from Cluster.clustermapper import ClusterMapper
  File "/mnt/temp/cloudtaf/libraries/Cluster/clustermapper.py", line 5, in <module>
    from Cluster.clusteragent import ClusterAgent
PYTHONPATH:
  /mnt/temp/cloudtaf/libraries
  /mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages
  /mnt/temp/rcp_cloudtaf_virtualenv/bin
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/plat-x86_64-linux-gnu
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-old
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-dynload
  /usr/lib/python2.7
  /usr/lib/python2.7/plat-x86_64-linux-gnu
  /usr/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/site-packages
  /mnt/temp/rcptaf
  /mnt/temp/rcptaf/libraries
  /mnt/temp/rcptaf/resources
  /mnt/temp/rcptaf/testcases/ELAM/libraries
[ ERROR ] Error in file '/mnt/temp/rcptaf/resources/rcp/base_platform/installation/installation_common.robot': Importing test library 'robot.libraries.String' failed: ImportError: cannot import name ascii_lowercase
Traceback (most recent call last):
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/String.py", line 18, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/string.py", line 18, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
PYTHONPATH:
  /mnt/temp/cloudtaf/libraries
  /mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages
  /mnt/temp/rcp_cloudtaf_virtualenv/bin
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/plat-x86_64-linux-gnu
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-old
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-dynload
  /usr/lib/python2.7
  /usr/lib/python2.7/plat-x86_64-linux-gnu
  /usr/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/site-packages
  /mnt/temp/rcptaf
  /mnt/temp/rcptaf/libraries
  /mnt/temp/rcptaf/resources
  /mnt/temp/rcptaf/testcases/ELAM/libraries
[ ERROR ] Error in file '/mnt/temp/rcptaf/resources/rcp/base_platform/installation/installation_common.robot': Importing test library 'robot.libraries.DateTime' failed: ImportError: cannot import name datetime
Traceback (most recent call last):
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/DateTime.py", line 303, in <module>
    from datetime import datetime, timedelta
  File "/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages/robot/libraries/datetime.py", line 303, in <module>
    from datetime import datetime, timedelta
PYTHONPATH:
  /mnt/temp/cloudtaf/libraries
  /mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages
  /mnt/temp/rcp_cloudtaf_virtualenv/bin
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/plat-x86_64-linux-gnu
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-old
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-dynload
  /usr/lib/python2.7
  /usr/lib/python2.7/plat-x86_64-linux-gnu
  /usr/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/site-packages
  /mnt/temp/rcptaf
  /mnt/temp/rcptaf/libraries
  /mnt/temp/rcptaf/resources
  /mnt/temp/rcptaf/testcases/ELAM/libraries
/mnt/temp/rcptaf/libraries/rcplib/connection/mgnt.py
[ ERROR ] Error in file '/mnt/temp/rcptaf/testcases': Importing test library 'RCPClusterMapper' failed: ImportError: No module named clusteragent
Traceback (most recent call last):
  File "/mnt/temp/rcptaf/libraries/RCPClusterMapper/__init__.py", line 1, in <module>
    from rcpclustermapper import RCPClusterMapper
  File "/mnt/temp/rcptaf/libraries/RCPClusterMapper/rcpclustermapper.py", line 6, in <module>
    from Cluster.clustermapper import ClusterMapper
  File "/mnt/temp/cloudtaf/libraries/Cluster/clustermapper.py", line 5, in <module>
    from Cluster.clusteragent import ClusterAgent
PYTHONPATH:
  /mnt/temp/cloudtaf/libraries
  /mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages
  /mnt/temp/rcp_cloudtaf_virtualenv/bin
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/plat-x86_64-linux-gnu
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-old
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/lib-dynload
  /usr/lib/python2.7
  /usr/lib/python2.7/plat-x86_64-linux-gnu
  /usr/lib/python2.7/lib-tk
  /mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/site-packages
  /mnt/temp/rcptaf
  /mnt/temp/rcptaf/libraries
  /mnt/temp/rcptaf/resources
  /mnt/temp/rcptaf/testcases/ELAM/libraries
  /mnt/temp/rcptaf/libraries

**Comment by HelioGuilherme66** [comment]
Please confirm your Robot Framework version. I see that you have two installs from your PYTHONPATH, the first found is `/mnt/temp/rcp_cloudtaf_virtualenv/local/lib/python2.7/site-packages` and the other not found or used is `/mnt/temp/rcp_cloudtaf_virtualenv/lib/python2.7/site-packages`.

**Comment by sliwinski-milosz** [comment]
I am able to reproduce that issue inside my vagrant machine if my repo directory is shared between the host and the guest.

Tested on:
```
Python 2.7.12
pip (9.0.1)
robotframework (3.0.2)
setuptools (38.2.4)
```

The reason why it is failing is in the traceback:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rf_tests/code/.venv/local/lib/python2.7/site-packages/robot/libraries/String.py", line 19, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
  File "/home/rf_tests/code/.venv/local/lib/python2.7/site-packages/robot/libraries/string.py", line 19, in <module>
    from string import ascii_lowercase, ascii_uppercase, digits
ImportError: cannot import name ascii_lowercase
```
It firstly goes to the `String.py` module of Robot Framework and then for some reason it goes to that RF `String` package again as a relative import while trying to import `string` Python built-in module (lowercase) -> and it fails because of circular import. Please take a look on paths in the traceback.

So the issue here is that it shouldn't go to the RF `String` module while we want Python built-in `string` one.

Please note that the same problem exists for other packages which have similar names to Python built-in ones like: DateTime:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rf_tests/code/.venv/local/lib/python2.7/site-packages/robot/libraries/DateTime.py", line 307, in <module>
    from datetime import datetime, timedelta
  File "/home/rf_tests/code/.venv/local/lib/python2.7/site-packages/robot/libraries/datetime.py", line 307, in <module>
    from datetime import datetime, timedelta
ImportError: cannot import name datetime
```

I am not able to reproduce this issue on Python 3.5.2 -> so seems like migration to the Python 3 is one way to fix the issue.

Another way would be to rename Robot Framework packages like: String, DateTime, Collections.

**Comment by pekkaklarck** [comment]
Interesting. Could you @sliwinski-milosz try adding `from __future__ import absolute_import` into the beginning of `String.py` before any other imports? Do you possibly have [PYTHONCASEOK](https://docs.python.org/2/using/cmdline.html#envvar-PYTHONCASEOK) environment variable set?

**Comment by sliwinski-milosz** [comment]
PYTHONCASEOK env variable is not set on my machine.

Please note that after I have added `from __future__ import absolute_import` on the top of `String.py`file - it works fine so I am able to do:
`from robot.libraries.String import ascii_lowercase`

I added that line to the `DateTime.py` file also and it fixed the `datetime` import issue as well:
`from robot.libraries.DateTime import datetime`

**Comment by pekkaklarck** [comment]
Just noticed that in the original description the error occurs already when starting Robot Framework when doing `import string`. That's because before RF 2.9 `robot/libraries` directory was added to `sys.path` automatically. Without the fix in #1737 adding `from future import absolute_import` wouldn't even work as `String` and friends would be directly importable anyway. Might be better to open a new issue but I guess we can live with this one too.

**Comment by pekkaklarck** [comment]
A nice side-effect of these fixes is that now Robot can be used with PYTHONCASEOK set. Setting it on before starting our acceptance test execution also validates that the fix itself works.

**Comment by sliwinski-milosz** [comment]
Thank you very much for fix!

Just out of the curiosity... Do you know what was the reason why it was reproducible while sharing directory and not if directory was not shared? Why import was working in the different way?

**Comment by pekkaklarck** [comment]
I don't know but guess it has something to do with (virtual) file system properties. If you read the earlier comments, you can see that the problem has also earlier happened when sharing folders between a VM and a host. Interestingly I've never encountered it although I have same setup using VirtualBox myself.

**Comment by pekkaklarck** [comment]
Noticed that `Remote` is affected as well. It tries to do `from xml.parsers.expat import ExpatError` which fails in the same conditions as discussed above. This import is surrounded by `try/except ImportError` due to IronPython compatibility and thus the problem wasn't so severe.
